### DISCUSSION TITLE
Change "Range" with the actual variables in the storage structs

### DISCUSSION THREAD

**Discussion by andreivladbrg** [sablier-labs/evm-monorepo#310] [General]
In this PR https://github.com/sablierhq/v2-core/pull/220 we faced some issues with the well known: "Stack too deep" error, and because of that we added a couple of [structs](https://github.com/sablierhq/v2-core/blob/prb/total-amount/src/types/DataTypes.sol) to get rid of some function parameters. One of them is the `Range` struct:

https://github.com/sablierhq/v2-core/blob/55a4d2ed225131247c97dd5ed352c44d755007eb/src/types/DataTypes.sol#L66-L70

Which has the purpose of replacing these variables from create function:
```solidity
    uint40 startTime; 
    uint40 cliffTime; 
    uint40 endTime; 
```

There is an issue with it: @PaulRBerg decided to also use it in the `storage` struct and the gas has increased with 21740 in `{SablierV2LockupLinear-createWithRange}`, the time variables are not tight packed anymore.
https://github.com/sablierhq/v2-core/blob/55a4d2ed225131247c97dd5ed352c44d755007eb/src/types/DataTypes.sol#L81-L88

<details>
<summary>Gas comparison</summary>

#### Linear

<img width="1353" alt="WithRangeLinear" src="https://user-images.githubusercontent.com/99738872/216351330-115f0ead-7f8b-4cf9-b163-e91973f656a6.png">

#### Pro

<img width="1356" alt="WithRangePro" src="https://user-images.githubusercontent.com/99738872/216351347-c925e09d-df71-4718-882e-26e5cb7443fe.png">

#### Linear

<img width="1332" alt="NoRangeLinear" src="https://user-images.githubusercontent.com/99738872/216351112-68812bed-073e-4739-85c8-aac6aaf5b073.png">

#### Pro

<img width="918" alt="NoRangePro" src="https://user-images.githubusercontent.com/99738872/216351013-47b5deb9-c82d-4192-9ab3-44954808c850.png">

</details>

My suggestion is to continue using the `Range` struct as a function parameter but to remove it from the storage struct. The downside is that the refactor will take time and I don't know if we can delay more, @razgraf needs the contracts for the app development.

**Comment by PaulRBerg** [comment]
Thank you for opening this discussion, Andrei.

I am sympathetic to your proposal, since 20k gas is non-negligible. Let me sleep more on this idea before we make a final decision - in the short term, we will pass on the contracts to the frontend team as they are now so they can start the refactor of the subgraph.

**Comment by PaulRBerg** [comment]
Locking, as we implemented this in https://github.com/sablierhq/v2-core/pull/334.
